src/Number.php

```diff
@@ -115,4 +115,42 @@ public static function fromFloat($floatingPoint)
 
         return new self(sprintf('%.8g', $floatingPoint));
     }
+
+    /**
+     * @return bool
+     */
+    public function isNegative()
+    {
+        return $this->number[0] === '-';
+    }
+
+    /**
+     * @return string
+     */
+    public function getIntegerPart()
+    {
+        if ($this->decimalSeparatorPosition === false) {
+            return $this->number;
+        }
+
+        return substr($this->number, 0, $this->decimalSeparatorPosition);
+    }
+
+    /**
+     * @return string
+     */
+    public function getFractionalPart()
+    {
+        if ($this->decimalSeparatorPosition === false) {
+            return '';
+        }
+
+        $fractionalPart = substr($this->number, $this->decimalSeparatorPosition + 1);
+
+        if (str_pad('', strlen($fractionalPart), '0') === $fractionalPart) {
```

Comment:
We can just do `return rtrim($fractionalPart, '0')` here, so it will be shorter and will also make the number `42.500` return the same fractional part as `42.5`.